File: app/src/main/java/in/sudopk/vaishnavacalendar/Country.java
package in.sudopk.vaishnavacalendar;

import java.util.ArrayList;
import java.util.List;

public class Country implements Comparable<Country> {
    private final String mName;
    private final List<Location> mLocations;

    public Country(final String name, final ArrayList<Location> locations) {
        mName = name;

        mLocations = locations;
    }

    public List<Location> getLocations() {
        return mLocations;
    }

    public String getName() {
        return mName;
    }

    @Override
    public int compareTo(final Country country) {
        return mName.compareTo(country.getName());
    }

    @Override
    public boolean equals(final Object obj) {
        return obj instanceof Country &&
                compareTo((Country) obj) == 0;
    }
}

File: app/src/main/java/com/sudopk/vc/calendar/Country.kt
package com.sudopk.vc.calendar

import com.sudopk.vc.location.Location

data class Country(val name: String, val locations: List<Location>) : Comparable<Country> {
  override fun compareTo(other: Country): Int {
    return name.compareTo(other.name)
  }

  override fun equals(other: Any?): Boolean {
    return other is Country && compareTo(other) == 0
  }

  override fun hashCode(): Int = name.hashCode()
}
